aw/Helpers/AWApiHelper.swift

```diff
@@ -0,0 +1,366 @@
+import CoreData
+import Foundation
+import MapKit
+import UIKit
+
+let baseURL = "https://www.americanwhitewater.org/content/"
+
+let riverURL = baseURL + "River/search/.json"
+
+struct AWApiHelper {
+    typealias ReachCallback = ([AWReach]?) -> Void
+    typealias UpdateCallback = () -> Void
+    typealias ReachDetailCallback = (AWReachDetailResponse.Sub.Main) -> Void
+
+    static func fetchReachesByRegion(region: String, callback: @escaping ReachCallback) {
+        let urlString = riverURL + "?state=\(region)"
+
+        guard let url = URL(string: urlString) else { return }
+
+        let task = URLSession.shared.dataTask(with: url) { data, response, error in
+            print("Data retrieved from AW API for \(region), decoding reaches")
+
+            let decoder = JSONDecoder()
+            guard let data = data, let reaches = try? decoder.decode([AWReach].self, from: data) else { return }
+
+            callback(reaches)
+        }
+        task.resume()
+    }
+
+    static func fetchReachesByIds(reachIds: [String], callback: @escaping ReachCallback) {
+        let urlString = baseURL + "River/list/list/\( reachIds.joined(separator: ":") )/.json"
+
+        guard let url = URL(string: urlString) else { return }
+
+        let task = URLSession.shared.dataTask(with: url) { data, response, error in
+            print("Data retrieved from AW API for \( reachIds ), decoding reaches")
+
+            let decoder = JSONDecoder()
+            guard let data = data, let reaches = try? decoder.decode([AWReach].self, from: data) else { return }
+
+            callback(reaches)
+        }
+        task.resume()
+    }
+
+    static func createOrUpdateReaches(newReaches: [AWReach], context: NSManagedObjectContext) {
```

Comment:
The use of the `NSManagedObjectContext` here is a little weird, and I think may cause some confusion in the future. If I'm reading this correctly, a consumer of this class fetches data directly from the api via `fetchReachesByRegion` and then is responsible for using this method to update Core Data (which is essentially just a cache). 
In general, it's clearer and less coupled if the consumer only interacts with one of, the source of truth, or the cache. There are two ways to do that here. First, you can hide the cache completely behind the api interface. Thus a consumer of the api wouldn't have to know about the cache at all, but would see the speed benefits. Second, you could have the consumer only interact with the cache, and the cache calls out to the api as necessary. I think 2) is better for you since you'll want to access other info from core data than just the api interface, but I'll leave it up to you.

Reply:
`createOrUpdateReaches` is only used within AWApiHelper by other static methods.

Reply:
Does this apply to all the rest of the methods in this file that use a managed context though?